### PR TITLE
[Tool] dockerfile enhancement

### LINKED ISSUE #30523: [allin1] remove be.conf/fe.conf from /data/deploy
## Enhancement

causes unnecessary confusion when browsing container filesystem content.

image

### PR DESCRIPTION
**Pull request by kevincai**
* remove unused fe.conf be.conf from allin1 docker image /data/deploy/ directory (Fixes #30523)
* tail be/cn out and info log file if not running with LOG_CONSOLE=1 in case of expect shutdown, give chances of last words to console for debugging

Fixes #issue

## What type of PR is this:

- [ ] BugFix
- [ ] Feature
- [X] Enhancement
- [ ] Refactor
- [ ] UT
- [ ] Doc
- [X] Tool

Does this PR entail a change in behavior?

- [ ] Yes, this PR will result in a change in behavior.
- [X] No, this PR will not result in a change in behavior.

If yes, please specify the type of change:

- [ ] Interface/UI changes: syntax, type conversion, expression evaluation, display information
- [ ] Parameter changes: default values, similar parameters but with different default values
- [ ] Policy changes: use new policy to replace old one, functionality automatically enabled
- [ ] Feature removed
- [ ] Miscellaneous: upgrade & downgrade compatibility, etc.

## Checklist:

- [ ] I have added test cases for my bug fix or my new feature
- [ ] This pr needs user documentation (for new or modified features or behaviors)
  - [ ] I have added documentation for my new feature or new function

## Bugfix cherry-pick branch check:

- [X] I have checked the version labels which the pr will be auto-backported to the target branch
  - [X] 3.2
  - [X] 3.1
  - [X] 3.0
  - [X] 2.5
  - [ ] 2.4

### CODE REVIEW

**Comment by wanpengfei-git** [comment]
#### [FE Incremental Coverage Report]


:heart_eyes: **pass** : 0 / 0 (0%)

**Comment by wanpengfei-git** [comment]
#### [BE Incremental Coverage Report]


:heart_eyes: **pass** : 0 / 0 (0%)

**Review by imay — APPROVED** [review]

**Review by sduzh — APPROVED** [review]

**Comment by kevincai** [comment]
@mergifyio backport branch-3.2 branch-3.1 branch-3.0 branch-2.5

File: docker/dockerfiles/be/be_entrypoint.sh [hunk 1]
```diff
@@ -115,3 +115,12 @@ if [[ "x$LOG_CONSOLE" == "x1" ]] ; then
     addition_args="--logconsole"
 fi
 $STARROCKS_HOME/bin/start_be.sh $addition_args
+ret=$?
+if [[ $ret -ne 0 && "x$LOG_CONSOLE" != "x1" ]] ; then
+    nol=50
+    log_stderr "Last $nol lines of be.INFO ..."
+    tail -n 50 $STARROCKS_HOME/log/be.INFO
```
Comment (yandongxiao) [inline docker/dockerfiles/be/be_entrypoint.sh:122]: tail -n $nol $STARROCKS_HOME/log/be.INFO

File: docker/dockerfiles/be/be_entrypoint.sh [hunk 2]
```diff
@@ -115,3 +115,12 @@ if [[ "x$LOG_CONSOLE" == "x1" ]] ; then
...
+    log_stderr "Last $nol lines of be.out ..."
+    tail -n 50 $STARROCKS_HOME/log/be.out
```
Comment (yandongxiao) [inline docker/dockerfiles/be/be_entrypoint.sh:124]: tail -n $nol $STARROCKS_HOME/log/be.out

File: docker/dockerfiles/be/cn_entrypoint.sh [hunk 3]
```diff
@@ -173,3 +173,12 @@ if [[ "x$LOG_CONSOLE" == "x1" ]] ; then
     addition_args="--logconsole"
 fi
 $STARROCKS_HOME/bin/start_cn.sh $addition_args
+ret=$?
+if [[ $ret -ne 0 && "x$LOG_CONSOLE" != "x1" ]] ; then
+    nol=50
+    log_stderr "Last $nol lines of cn.INFO ..."
+    tail -n $nol $STARROCKS_HOME/log/cn.INFO
+    log_stderr "Last $nol lines of cn.out ..."
+    tail -n $nol $STARROCKS_HOME/log/cn.out
```
Comment (yandongxiao) [inline docker/dockerfiles/be/cn_entrypoint.sh:182]: I checked a k8s env, and there is no `$STARROCKS_HOME` environment.

Comment (kevincai) [reply docker/dockerfiles/be/cn_entrypoint.sh:182]: it is set in entrypoint

File: docker/dockerfiles/be/be_entrypoint.sh (hunk 2 above)
Comment (kevincai) [reply docker/dockerfiles/be/be_entrypoint.sh:124]: good point

**Pull request merged**